.github/workflows/utils.sh

```diff
@@ -152,15 +154,214 @@ function unzipArtifact(){
+  cd src/sdks/core
+  npm run cpp
+
+  if [ $? -eq 0 ]
+  then
+        echo "Native Core SDK generated successfully"
+        echo " Core SDK Location"
+        cd build/cpp/src/
+        ls -la
+        echo " ************ Source Generation Completed for Core SDK ************"
+  else
+    echo "Native Core SDK generation failed"
+    exit 1
+  fi
+
+  echo "Generating source for Manage SDK"
+  cd ../../../../manage
+  npm run cpp
+
+  if [ $? -eq 0 ]
+  then
+        echo "Native Manage SDK generated successfully"
+        echo " Manage SDK Location"
+        cd build/cpp/src/
+        ls -la
+        echo " ************ Source Generation Completed for Manage SDK ************"
+  else
+    echo "Native Manage SDK generation failed"
+    exit 1
+  fi
+
+  echo "Generate source for Discovery SDK"
+  cd ../discovery
+  npm run cpp
+
+  if [ $? -eq 0 ]
+  then
+        echo "Native Discovery SDK generated successfully"
+  else
+    echo "Native Discovery SDK generation failed"
+    exit 1
+  fi
+}
+
+function cloneAndInstallThunder() {
+  cd ..
+
+  git clone https://github.com/rdkcentral/Thunder.git
+
+  cd Thunder
+
+  git checkout 283b3d54334010403d85a4e69b3835de23e42331
+
+  cd ..
+
+  git clone https://github.com/rdkcentral/ThunderTools.git
+
+  cd ThunderTools
+
+  git checkout 64b72b5ed491436b0e6bc2327d8a7b0e75ee2870
+
+  cd ..
+
+  echo "current_dir is $current_dir"
+
+  mkdir -p $current_dir/data
+
+  if [ ! -d "$(pwd)/Thunder" ] 
+  then
+      echo "Directory Thunder DOES NOT exist." 
+      exit 9999
+  fi
+
+  if [ ! -d "$(pwd)/ThunderTools" ]
+  then
+      echo "Directory ThunderTools DOES NOT exist."
+      exit 9999
+  fi
+
+  if [ ! -d "$current_dir/firebolt-apis" ]
+  then
+      echo "Directory $current_dir/firebolt-apis DOES NOT exist."
+      exit 9999
+  fi
+
+  cd $current_dir
+
+  [ -d "$current_dir/build" ] && rm -rf $current_dir/build
+
+  echo "Building Thunder";
+
+  cmake -G Ninja -S ThunderTools -B build/ThunderTools -DCMAKE_INSTALL_PREFIX="install/usr" && echo "Tools Setup" || exit 9999
+
+  cmake --build build/ThunderTools --target install && echo "Thunder Tools Build succeeded" || exit 9999
+
+  #-G Ninja is the "Build system generator"
+  #-S is the source path
+  #-B is the output directory
+  cmake -G Ninja -S Thunder -B build/Thunder \
+  -DBUILD_SHARED_LIBS=ON \
+  -DBINDING="127.0.0.1" \
+  -DCMAKE_BUILD_TYPE="Debug" \
+  -DCMAKE_INSTALL_PREFIX="install/usr" \
+  -DCMAKE_MODULE_PATH="${PWD}/install/usr/include/WPEFramework/Modules" \
+  -DDATA_PATH="${PWD}/install/usr/share/WPEFramework" \
+  -DPERSISTENT_PATH="${PWD}/install/var/wpeframework" \
+  -DPORT="55555" \
+  -DPROXYSTUB_PATH="${PWD}/install/usr/lib/wpeframework/proxystubs" \
+  -DSYSTEM_PATH="${PWD}/install/usr/lib/wpeframework/plugins" \
+  -DVOLATILE_PATH="tmp" && echo "Thunder configure succeeded" || exit 9999
+
+  cmake --build build/Thunder --target install && echo "Thunder Build succeeded" || exit 9999
+}
+
+function buildCoreCPPSDK() {
```

Comment:
Can we combine the 3 build functions into one function that takes in an SDK parameter? We'll have other SDKs to worry about down the road and it'll be easier to add it as a parameter.

Reply:
I also don't see these functions being used anywhere.